tests/filecheck/dialects/riscv/riscv_ops_custom_syntax.mlir

```diff
@@ -0,0 +1,270 @@
+  // CHECK-NEXT: %xori = riscv.xori %0, 1 : (!riscv.reg<>) -> (!riscv.reg<>)
+  // CHECK-NEXT: %slli = riscv.slli %0, 1 : (!riscv.reg<>) -> (!riscv.reg<>)
+  // CHECK-NEXT: %srli = riscv.srli %0, 1 : (!riscv.reg<>) -> (!riscv.reg<>)
+  // CHECK-NEXT: %srai = riscv.srai %0, 1 : (!riscv.reg<>) -> (!riscv.reg<>)
+  // CHECK-NEXT: %lui = riscv.lui 1 : () -> (!riscv.reg<>)
+  // CHECK-NEXT: %auipc = riscv.auipc 1 : () -> (!riscv.reg<>)
+  // CHECK-NEXT: %mv = riscv.mv %0 : (!riscv.reg<>) -> (!riscv.reg<>)
+
+  // Integer Register-Register Operations
+  %add = "riscv.add"(%0, %1) : (!riscv.reg<>, !riscv.reg<>) -> !riscv.reg<>
+  %slt = "riscv.slt"(%0, %1) : (!riscv.reg<>, !riscv.reg<>) -> !riscv.reg<>
+  %sltu = "riscv.sltu"(%0, %1) : (!riscv.reg<>, !riscv.reg<>) -> !riscv.reg<>
+  %and = "riscv.and"(%0, %1) : (!riscv.reg<>, !riscv.reg<>) -> !riscv.reg<>
+  %or = "riscv.or"(%0, %1) : (!riscv.reg<>, !riscv.reg<>) -> !riscv.reg<>
+  %xor = "riscv.xor"(%0, %1) : (!riscv.reg<>, !riscv.reg<>) -> !riscv.reg<>
+  %sll = "riscv.sll"(%0, %1) : (!riscv.reg<>, !riscv.reg<>) -> !riscv.reg<>
+  %srl = "riscv.srl"(%0, %1) : (!riscv.reg<>, !riscv.reg<>) -> !riscv.reg<>
+  %sub = "riscv.sub"(%0, %1) : (!riscv.reg<>, !riscv.reg<>) -> !riscv.reg<>
+  %sra = "riscv.sra"(%0, %1) : (!riscv.reg<>, !riscv.reg<>) -> !riscv.reg<>
+  "riscv.nop"() : () -> ()
+  // CHECK-NEXT: %add = riscv.add %0, %1 : (!riscv.reg<>, !riscv.reg<>) -> (!riscv.reg<>)
+  // CHECK-NEXT: %slt = riscv.slt %0, %1 : (!riscv.reg<>, !riscv.reg<>) -> (!riscv.reg<>)
+  // CHECK-NEXT: %sltu = riscv.sltu %0, %1 : (!riscv.reg<>, !riscv.reg<>) -> (!riscv.reg<>)
+  // CHECK-NEXT: %and = riscv.and %0, %1 : (!riscv.reg<>, !riscv.reg<>) -> (!riscv.reg<>)
+  // CHECK-NEXT: %or = riscv.or %0, %1 : (!riscv.reg<>, !riscv.reg<>) -> (!riscv.reg<>)
+  // CHECK-NEXT: %xor = riscv.xor %0, %1 : (!riscv.reg<>, !riscv.reg<>) -> (!riscv.reg<>)
+  // CHECK-NEXT: %sll = riscv.sll %0, %1 : (!riscv.reg<>, !riscv.reg<>) -> (!riscv.reg<>)
+  // CHECK-NEXT: %srl = riscv.srl %0, %1 : (!riscv.reg<>, !riscv.reg<>) -> (!riscv.reg<>)
+  // CHECK-NEXT: %sub = riscv.sub %0, %1 : (!riscv.reg<>, !riscv.reg<>) -> (!riscv.reg<>)
+  // CHECK-NEXT: %sra = riscv.sra %0, %1 : (!riscv.reg<>, !riscv.reg<>) -> (!riscv.reg<>)
+  // CHECK-NEXT: riscv.nop : () -> ()
+
+  // RV32I/RV64I: 2.5 Control Transfer Instructions
+  // terminators continue at the end of module
+
+  // Unconditional Branch Instructions
+  "riscv.jal"() {"immediate" = 1 : i32} : () -> ()
+  "riscv.jal"() {"immediate" = 1 : i32, "rd" = !riscv.reg<>} : () -> ()
+  "riscv.jal"() {"immediate" = #riscv.label<"label">} : () -> ()
+  // CHECK-NEXT: riscv.jal 1 : () -> ()
+  // CHECK-NEXT: riscv.jal 1, None : () -> ()
+  // CHECK-NEXT: riscv.jal "label" : () -> ()
+
+  "riscv.j"() {"immediate" = 1 : i32} : () -> ()
+  "riscv.j"() {"immediate" = #riscv.label<"label">} : () -> ()
+  // CHECK-NEXT: riscv.j 1 : () -> ()
+  // CHECK-NEXT: riscv.j "label" : () -> ()
```

Comment:
If you want, you can omit operation signatures if they are always empty (operations such as `j` will never have ssa arguments or return values, there we don't need to print them).

We can add this in a later PR too, we don't need to do everything in one major effort

Reply:
Sure, this can be put into future PR.

Reply:
I disagree, if we're doing a custom syntax then let's try to get it as close as possible to the goal.